app/controllers/actions/IdentifierAction.scala

```diff
@@ -0,0 +1,59 @@
+package controllers.actions
+
+import com.google.inject.Inject
+import config.FrontendAppConfig
+import controllers.routes
+import models.requests.IdentifierRequest
+import play.api.mvc.Results._
+import play.api.mvc._
+import uk.gov.hmrc.auth.core._
+import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals
+import uk.gov.hmrc.http.{HeaderCarrier, UnauthorizedException}
+import uk.gov.hmrc.play.HeaderCarrierConverter
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait IdentifierAction extends ActionBuilder[IdentifierRequest, AnyContent] with ActionFunction[Request, IdentifierRequest]
+
+class AuthenticatedIdentifierAction @Inject()(
+                                               override val authConnector: AuthConnector,
+                                               config: FrontendAppConfig,
+                                               val parser: BodyParsers.Default
+                                             )
+                                             (implicit val executionContext: ExecutionContext) extends IdentifierAction with AuthorisedFunctions {
+
+  override def invokeBlock[A](request: Request[A], block: IdentifierRequest[A] => Future[Result]): Future[Result] = {
+
+    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
+
+    authorised().retrieve(Retrievals.internalId) {
```

Comment:
do we need this? it's going to be an unauthenticated journey.

Reply:
yeah, its mostly auto generated, i can delete it now or in the upcoming PRs as we go along

Reply:
best to get rid off it now